eng/tools/generator/cmd/v2/automation/automationCmd.go

```diff
@@ -101,64 +102,136 @@ func (ctx *automationContext) generate(input *pipeline.GenerateInput) (*pipeline
 		return nil, fmt.Errorf("failed to get sdk repo: %+v", err)
 	}
 
-	if input.RelatedReadmeMdFile != "" {
-		input.RelatedReadmeMdFiles = append(input.RelatedReadmeMdFiles, input.RelatedReadmeMdFile)
-	}
+	// typespec
+	// Generated by tsp only when tspconfig.yaml exists and has typespec-go option
+	existTypeSpecProject := false
+	if len(input.RelatedTypeSpecProjectFolder) == 0 {
+		for _, tspProjectFolder := range input.RelatedTypeSpecProjectFolder {
+			tspconfigPath := filepath.Join(tspProjectFolder, "tspconfig.yaml")
+			tsc, err := typespec.ParseTypeSpecConfig(tspconfigPath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse tspconfig.yaml: %+v", err)
+			}
 
-	for _, readme := range input.RelatedReadmeMdFiles {
-		log.Printf("Start to process readme file: %s", readme)
+			if _, ok := tsc.TypeSpecProjectSchema.Options[string(typespec.TypeSpec_GO)]; ok {
+				existTypeSpecProject = true
+				generateCtx := common.GenerateContext{
```

Comment:
Can you log a message like autorest? 

'log.Printf("Start to process typespec project: %s", tspProjectFolder)'